tgz.go

```diff
@@ -20,12 +24,24 @@ type ArchiveWriter struct {
 	Writer  *bytes.Buffer
 }
 
+type ArchiveWriteTask struct {
+	File    string
+	Content []byte
+}
+
 func (a *ArchiveWriter) Write(p []byte) (n int, err error) {
 	return a.Writer.Write(p)
 }
 
 func (a *ArchiveWriter) Close() error {
-	return a.Archive.AddFileByContent(a.Writer.Bytes(), a.File)
+	if len(strings.Trim(a.File, "	 ")) == 0 {
```

Comment:
why do we need to trim like this. Are you just trying to remove spaces? you can use

strings.TrimSpace()